### PR TITLE
Merge template_v2.1.0

### PR DESCRIPTION
**Pull request by szczys**
This merges the upstream template_v2.1.0,

As part of the update, app_settings was overhauled to better mesh with the updated API.

Following the merge the app overflows the available flash space. The app_state was refactored to reduce binary size.

resolves https://github.com/golioth/devrel-issue-tracker/issues/387

### CODE REVIEW

**Review by cdwilson — COMMENTED** [review]
FYI, some additional feedback for `app_state` coming in an additional PR against the `szczys/merge_template_v2.1.0` branch.

**Comment by cdwilson** [comment]
FYI, some additional feedback in https://github.com/golioth/thingy91-golioth/pull/9 (intended to be merged into this PR and squashed).

File: src/app_settings.c [hunk 1]
```diff
@@ -93,160 +103,131 @@ int32_t get_loop_delay_s(void)
...
 
-	if (err) {
-		LOG_ERR("Failed to observe settings: %d", err);
-	}
-	return err;
+	LOG_ERR("Failed to register settings callback for %s: %d", settings_str, err);
 }
 
-int app_settings_register(struct golioth_client *settings_client)
+int app_settings_register(struct golioth_client *client)
```
[the remark this answers is not in the document]
Comment (szczys) [reply src/app_settings.c:181]: It's a good callout. The obvious issue is that you can't do `client = client` so the param name needs to be something different when passing it from main to the other C files that need it.

**Review by cdwilson — APPROVED** [review]
LGTM

**Pull request merged**